src/components/projects/wbs-details/work-package-container/work-package-container.tsx

```diff
@@ -30,24 +35,105 @@ export interface EditModeProps {
 const WorkPackageContainer: React.FC<WorkPackageContainerProps> = ({ wbsNum }) => {
   const { isLoading, isError, data, error } = useSingleWorkPackage(wbsNum);
   const [editMode, setEditMode] = useState(false);
+  const [form, setForm] = useState({});
+  const [expectedActivities, setExpectedActivities] = useState(['']);
+  const [deliverables, setDeliverables] = useState(['']);
+
+  // On render, data is undefined. This causes issues if trying to set the state to fields from it. Bug?
```

Comment:
Should I do anything else about this?

Reply:
I think how I resolved this is to make a wrapper component that gets the data separately and then passes that data to the view component (I presume that is this). That way, the view component will be guaranteed to always have data that is defined such that you show the loading indicator while you wait for the data, and then once the data is retrieved, you render the form with the data at the same time.

Reply:
Is it alright if I keep it like this? It works, but I'm also a bit confused about how I could implement such a wrapper. Should I create a component up or down the tree?

Reply:
If you went with a wrapper component, you'd make a component up the tree. It would do all the data processing and contain all the business logic like handler functions and things of that sort, and then pass that all down to another component (which I believe is what this is) that acts as a view. There are quite a few examples of this pattern throughout the codebase -- a simple example of this would be the review change request form if you want to check that out.